docs/syntax-declaration.md

```diff
@@ -0,0 +1,105 @@
+# Declaration syntax
+
+## Summary
+
+Formalizing the Luau declaration syntax used in the Luau parser.
+
+## Motivation
+
+RIght now the declaration syntax used by the Luau parser is undocumented and an unstable implementation detail. By defining the syntax, it allows for embedders of Luau to have typechecking for their apis without having to resort to empty modules. This should allow tools like luau-lsp to be able to provide easier ways to use these declarations.
```

Comment:
luau-lsp already uses declaration files as they are, so it's not really clear what we hope to actually gain here. the syntax of declaration files has been essentially unchanged since they were introduced, and it seems like the only thing this RFC has actually done is _inspire_ conversation to make changes to it.